integration-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,116 @@
+#![deny(missing_docs)]
+
+use std::path::PathBuf;
+use std::process::Command;
+use std::time::Duration;
+use wait_timeout::ChildExt;
+
+#[cfg_attr(not(any(has_sgx, has_sev)), ignore)]
+pub fn keep_under_test() -> &'static str {
+    let mut result = "";
+    if cfg!(has_sgx) {
+        result = "enarx-keep-sgx";
+    }
+    if cfg!(has_sev) {
+        result = "enarx-keep-sev";
+    }
+    result
+}
+
+pub struct IntegrationTest {
+    pub wksp_root: PathBuf,
+    pub keep: PathBuf,
+    pub payload: PathBuf,
+    pub shim: PathBuf,
+}
+
+impl IntegrationTest {
+    pub fn new(bin: &str) -> Self {
+        let enarx_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let enarx_dir = enarx_dir.parent().unwrap();
+        let keep_type = keep_under_test();
+        Self {
+            wksp_root: enarx_dir.to_path_buf(),
+            keep: find_filepath(
+                enarx_dir
+                    .join(format!("{}/target/", keep_type))
+                    .to_path_buf(),
+                format!("{}", keep_type).as_str(),
+            )
+            .unwrap(),
+            payload: find_filepath(
+                enarx_dir
+                    .join("integration-tests/target/x86_64-unknown-linux-musl/")
+                    .to_path_buf(),
+                bin,
+            )
+            .unwrap(),
+            shim: find_filepath(
+                enarx_dir
+                    .join(format!(
+                        "{}-shim/target/x86_64-unknown-linux-musl/",
+                        keep_type
+                    ))
+                    .to_path_buf(),
+                format!("{}-shim", keep_type).as_str(),
+            )
+            .unwrap(),
+        }
+    }
+    pub fn run_test(
+        self: Self,
+        timeout: Option<usize>,
+        exit_status: Option<i32>,
+        _stdin: Option<&str>,
+        _stdout: Option<&str>,
+        _stderr: Option<&str>,
+    ) -> () {
+        let mut cmd = Command::new(self.keep)
+            .current_dir(self.wksp_root)
+            .arg("--code")
+            .arg(self.payload)
+            .arg("--shim")
+            .arg(self.shim)
+            .spawn()
+            .expect("failed to run the test");
+        let num_sec = match timeout {
+            Some(num_sec) => num_sec,
+            None => 5,
+        };
+        let exit_status = match exit_status {
+            Some(exit_status) => exit_status,
+            None => 0,
+        };
+        let ecode = match cmd
+            .wait_timeout(Duration::from_secs(num_sec as u64))
+            .unwrap()
+        {
+            Some(status) => status.code(),
+            None => {
+                cmd.kill().unwrap();
+                panic!("killed by watchdog!");
+            }
+        };
+        assert_eq!(exit_status, ecode.unwrap());
+    }
+}
+
+/// Finds an absolute file path for a file in a directory.
+#[cfg_attr(not(has_sgx), ignore)]
+pub fn find_filepath(dir: PathBuf, name: &str) -> Result<PathBuf, std::io::Error> {
```

Comment:
```suggestion
pub fn find_filepath(dir: impl AsRef<Path>, name: &str) -> Result<PathBuf, std::io::Error> {
```

You can see the same strategy [here](https://doc.rust-lang.org/std/fs/fn.read.html).

Basically, any time you're going to take a path as an input, you want to instead take "any type that can be expressed as a `Path`." This allows you to use things like string literals.